src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
+import {
+  FC,
+  FormEvent,
+  ChangeEvent,
+  useState,
+} from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserByID(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserByID(todo.userId),
+}));
+
+export const App: FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [title, setTitle] = useState('');
+  const [usersValue, setUsersValue] = useState('0');
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUsers, setErrorUsers] = useState(false);
+
+  const handleChangeTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setErrorTitle(false);
+    setTitle(event.target.value);
+  };
+
+  const handleChangeUsers = (event: ChangeEvent<HTMLSelectElement>) => {
+    setErrorUsers(false);
+    setUsersValue(event.target.value);
+  };
+
+  const user = getUserByID(+usersValue);
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    const trimmedTitle = title.trim();
+
+    setErrorTitle(!trimmedTitle);
```

Comment:
set error if you don't have a value
```suggestion
    if (! trimmedTitle) {
      setIsErrorTitle(true);
    }
```

Reply:
It's the same, no?